lib/components/src/addon-panel/addon-panel.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useRef, useEffect, ReactNode } from 'react';
+
+const usePrevious = (value: any) => {
+  const ref = useRef();
+
+  useEffect(() => {
+    // happens after return
+    ref.current = value;
+  }, [value]);
+
+  return ref.current;
+};
+
+const useUpdate = (update: boolean, value: any) => {
```

Comment:
Comment these? When/how are they supposed to be used?

Reply:
it's not exported, so it's just for use in this one place for now.